tests/Microsoft.Bot.Connector.Tests/OAuthConnectorTests.cs

```diff
@@ -59,45 +86,241 @@ public async Task GetUserToken_ShouldReturnNullOnInvalidConnectionString()
+        {
+            await Assert.ThrowsAsync<ArgumentNullException>(() => OAuthClientConfig.SendEmulateOAuthCardsAsync(
+                null, true));
+        }
+
+        [Fact]
+        public async Task SendEmulateOAuthCardsAsync_ShouldThrowOnNoLocalBot()
+        {
+            var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
+            ServiceClientTracing.IsEnabled = true;
+
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("AGENT_OS")) &&
+                Environment.GetEnvironmentVariable("AGENT_OS").Equals("Windows_NT"))
+            {
+                // Automated Windows build exception:
+                await OAuthClientConfig.SendEmulateOAuthCardsAsync(client, true);
+                Assert.True(true, "No exception thrown.");
+            }
+            else
+            {
+                // MacLinux build and local build exception:
+                await Assert.ThrowsAsync<System.Net.Http.HttpRequestException>(() => OAuthClientConfig.SendEmulateOAuthCardsAsync(
+                    client, true));
+            }
+        }
+
+        [Fact]
+        public async Task GetSignInResourceWithHttpMessagesAsync_ShouldThrowOnNullState()
+        {
+            var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
+            await Assert.ThrowsAsync<ValidationException>(() => client.GetSignInResourceAsync(
+                null, null));
+        }
+
+        [Fact]
+        public async Task GetSignInResourceWithHttpMessagesAsync_ShouldThrowOnNoLocalBot()
+        {
+            var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
+            ServiceClientTracing.IsEnabled = true;
+
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("AGENT_OS")) &&
+                Environment.GetEnvironmentVariable("AGENT_OS").Equals("Windows_NT"))
+            {
+                // Automated Windows build exception:
+                await Assert.ThrowsAsync<Microsoft.Rest.HttpOperationException>(() => client.GetSignInResourceAsync(
+                    "dummyState", "dummyCodeChallenge", "dummyEmulatorUrl", "dummyFinalRedirect"));
+            }
+            else
+            {
+                // MacLinux build and local build exception:
+                await Assert.ThrowsAsync<System.Net.Http.HttpRequestException>(() => client.GetSignInResourceAsync(
+                    "dummyState", "dummyCodeChallenge", "dummyEmulatorUrl", "dummyFinalRedirect"));
+            }
+        }
+
+        [Fact]
+        public async Task ExchangeAsyncWithHttpMessagesAsync_ShouldThrowOnNullUserId()
+        {
+            var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
+            await Assert.ThrowsAsync<ValidationException>(() => client.ExchangeAsyncAsync(
+                null, null, null, null));
```

Comment:
I think the only null value for this test should be the UserId to make sure the method is throwing for the expected argument.